### PR TITLE
remove c_nonce and c_nonce_expires_in from credential endpoint response

### LINKED ISSUE #393: remove the option to return c_nonce from the credential response
This only makes sense when/if/after #381 is merged. 

_Originally posted by @bc-pi in relation to https://github.com/openid/OpenID4VCI/issues/381#issuecomment-2357047069_

> > Discussed on the WG call, direction on removing section 7.3.2 and consolidating its content in Nonce Endpoint definition. https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-wg-draft.html#section-7.3.2

> While I was part of that discussion, after looking again at that section and content of this PR, I don't see how that consolation could actually manifest in a meaningful way. I suggest this PR proceed without and consider that as part of the soon to be separate issue on removing an option to return c_nonce from the credential [error] response.

 > > also agreed to open a separate issue on removing an option to return c_nonce from the credential error response, so that we can merge this PR as-is (@bc-pi).

that is this

### PR DESCRIPTION
**Pull request by mickrau**
resolves #393

Simplify spec by only provide one way to get a c_nonce

### CODE REVIEW

**Review by paulbastian — APPROVED** [review]

**Review by c2bo — COMMENTED** [review]
This PR provides a nice simplification but from my understanding has 2 potential downsides:
- adds a round trip in the corner cases where a server rejects a proof because the nonce was deemed invalid
- removes the possibility to provide stateful nonces - the nonce endpoint is not protected and there is no possibility (I think?) for this right now

The only thing that worries me a bit is losing the possibility to provide stateful nonces, but I am not entirely sure if we need them - are people using that right now? If we really want/need to preserve the ability to have stateful nonces, then we could also add the option to have the nonce endpoint protected and signal that via metadata I guess?

File: openid-4-verifiable-credential-issuance-1_0.md
Comment (paulbastian) [reply openid-4-verifiable-credential-issuance-1_0.md:1095]: agree very much, distinguishing this error, makes it clear for the wallet what to do

Comment (mickrau) [reply openid-4-verifiable-credential-issuance-1_0.md]: Thanks @c2bo, i also thought about a specific error code and like your suggestion

File: openid-4-verifiable-credential-issuance-1_0.md
Comment (nemqe) [inline openid-4-verifiable-credential-issuance-1_0.md:1096]: If we have the `invalid nonce` error would it make sense to use it also for the `invalid_proof` case (4)?

**Review by nemqe — APPROVED** [review]

**Comment by nemqe** [comment]
If I see it correctly this PR also addresses https://github.com/openid/OpenID4VCI/issues/394 right?

**Comment by Sakurann** [comment]
with my chair hat on, noting that there is not enough discussion in the issue to be called an agreement to remove this feature.

**Comment by paulbastian** [comment]
Neither was there enough discussion on this in the none endpoint PR, but Brian just said he doesn't see this. I think the PR helps to make people understand this and reflects our real developer experience from implementing nonce endpoint.

**Comment by Sakurann** [comment]
Brian first did a PR without discussion and that one got closed, we did discuss that in the following calls and there are multiple other issuer where the topic was discussed so when the PR came up the second time, the WG was more prepared.

**Comment by bc-pi** [comment]
> If I see it correctly this PR also addresses #394, right?

No, the `## Nonce Response {#nonce-response}` section  still has a `c_nonce_expires_in` so this one doesn't fully address #394.

https://github.com/openid/OpenID4VCI/pull/404/files#diff-1f424614b35a9899813079f1b1f6218631a2aedd993368ccb89bb81a9eda0289R755

**Comment by bc-pi** [comment]
> Neither was there enough discussion on this in the none endpoint PR, but Brian just said he doesn't see this.

I'm not entirely sure I know what that means. But there were hundreds of comments across #381, #199, & #39 discussing removal of c_nonce* from the AS's token endpoint. The issue behind this PR #393 and an issue to get rid of c_nonce_expires_in in general #394 were broken out from #381 in an attempt to keep its scope tight and allow it to proceed. The nonce endpoint was added to #381, which was an expansion of scope, but as a condition of acceptance by a few key opponents. And even that had quite a bit of discussion.

**Comment by bc-pi** [comment]
> This PR provides a nice simplification but from my understanding has 2 potential downsides:
> 
>     * adds a round trip in the corner cases where a server rejects a proof because the nonce was deemed invalid
> 
>     * removes the possibility to provide stateful nonces - the nonce endpoint is not protected and there is no possibility (I think?) for this right now
> 
> 
> The only thing that worries me a bit is losing the possibility to provide stateful nonces, but I am not entirely sure if we need them - are people using that right now? 

Stateful with respect to the access token or something contained therein. But good question. 

>If we really want/need to preserve the ability to have stateful nonces, then we could also add the option to have the nonce endpoint protected and signal that via metadata I guess?

That could be signaled by a 401 at the nonce endpoint too.

File: openid-4-verifiable-credential-issuance-1_0.md
Comment (nemqe) [inline openid-4-verifiable-credential-issuance-1_0.md:1095]:
```suggestion
  * `invalid_proof`: The `proof` or `proofs` parameter in the Credential Request is invalid: (1) if both fields are missing, or (2) both are present simultaneously, or (3) one of the provided key proofs is invalid, or (4) if at least one of the key proofs does not contain a `c_nonce` value (refer to (#nonce-response)).
```

File: openid-4-verifiable-credential-issuance-1_0.md
Comment (nemqe) [inline openid-4-verifiable-credential-issuance-1_0.md:1096]:
```suggestion
  * `invalid_nonce`: The `proof` or `proofs` parameter in the Credential Request uses an invalid nonce: at least one of the key proofs does not contain a valid `c_nonce` value. The wallet should retrieve a new `c_nonce` value (refer to (#nonce-endpoint)).
```

**Review by bc-pi — APPROVED** [review]

**Comment by Sakurann** [comment]
agreed to proceed with this PR, merging once 1-2 more approvals

**Review by jogu — APPROVED** [review]

**Review by peppelinux — APPROVED** [review]

**Review by tlodderstedt — APPROVED** [review]

**Comment by Sakurann** [comment]
@mickrau can you please add a changelog entry that this option was removed? I will merge this PR after that

**Review by c2bo — APPROVED** [review]

**Comment by mickrau** [comment]
> @mickrau can you please add a changelog entry that this option was removed? I will merge this PR after that

@Sakurann  done

**Review by Sakurann — APPROVED** [review]

**Pull request merged**
